metals/src/main/scala/scala/meta/internal/metals/callHierarchy/RealRoot.scala

```diff
@@ -0,0 +1,23 @@
+package scala.meta.internal.metals.callHierarchy
+
+import scala.meta.Tree
+import scala.meta.Name
+
+/**
+ * Where to start looking for occurrences.
+ * When using the findLastEnclosingAt method, it often happens that you get Name, not the root of the definition,
+ * this trait allows to store the root and the potential name of a definition.
+ */
+private[callHierarchy] sealed trait RealRoot {
+  def root: Tree
+  def mayName: Option[Name]
```

Comment:
I find `mayName` a bit confusing. What if instead we just made this:

```
name: Option[Name]
```
And then change `NamedRealRoot` to

```
private[callHierarchy] case class NamedRealRoot(root: Tree, private val _name: Name)
    extends RealRoot {
  val name = Some(name)
}
```

So then you'd just always try to access `name` which is either there or not.